src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java

```diff
@@ -229,7 +230,9 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     }
     // try evaluating the script to fail application creation
     // if the script is invalid (only possible for interpreted mode)
-    if (config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) {
+    if (config.executionMode != null &&
```

Comment:
`!containsMacro(config.executionMode)`

We should also throw validation exception if `executionMode` is `null`:
```
if (!containsMacro(config.executionMode) && Strings.isNullOrEmpty(config.executionMode)) {
.....
}
```

Reply:
Updated !